src/main/java/com/example/weather/client/services/WeatherDataService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.weather.client.services;
+
+import com.example.weather.client.client.WeatherClient;
+import com.example.weather.client.mappers.WeatherDataMapper;
+import com.example.weather.client.repositories.WeatherDataRepo;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+
+@Setter
+@Service
+@RequiredArgsConstructor
+@EnableAsync
```

Comment:
a better practice is to keep all `@Enable~`s in one generic place, so move this annotation to Application.